jupyter/d2l-java/chapter_linear-networks/linear-regression.ipynb

```diff
@@ -0,0 +1,871 @@
+    "\n",
+    "$$p(y|\\mathbf{x}) = \\frac{1}{\\sqrt{2 \\pi \\sigma^2}} \\exp\\left(-\\frac{1}{2 \\sigma^2} (y - \\mathbf{w}^\\top \\mathbf{x} - b)^2\\right).$$\n",
+    "\n",
+    "Now, according to the *maximum likelihood principle*,\n",
+    "the best values of $b$ and $\\mathbf{w}$ are those\n",
+    "that maximize the *likelihood* of the entire dataset:\n",
+    "\n",
+    "$$P(Y\\mid X) = \\prod_{i=1}^{n} p(y^{(i)}|\\mathbf{x}^{(i)}).$$\n",
+    "\n",
+    "Estimators chosen according to the *maximum likelihood principle*\n",
+    "are called *Maximum Likelihood Estimators* (MLE).\n",
+    "While, maximizing the product of many exponential functions,\n",
+    "might look difficult,\n",
+    "we can simplify things significantly, without changing the objective,\n",
+    "by maximizing the log of the likelihood instead.\n",
+    "For historical reasons, optimizations are more often expressed\n",
+    "as minimization rather than maximization.\n",
+    "So, without changing anything we can minimize the *Negative Log-Likelihood (NLL)*\n",
+    "$-\\log p(\\mathbf y|\\mathbf X)$.\n",
+    "Working out the math gives us:\n",
+    "\n",
+    "$$-\\log p(\\mathbf y|\\mathbf X) = \\sum_{i=1}^n \\frac{1}{2} \\log(2 \\pi \\sigma^2) + \\frac{1}{2 \\sigma^2} \\left(y^{(i)} - \\mathbf{w}^\\top \\mathbf{x}^{(i)} - b\\right)^2.$$\n",
+    "\n",
+    "Now we just need one more assumption: that $\\sigma$ is some fixed constant.\n",
+    "Thus we can ignore the first term because\n",
+    "it does not depend on $\\mathbf{w}$ or $b$.\n",
+    "Now the second term is identical to the squared error objective introduced earlier,\n",
+    "but for the multiplicative constant $\\frac{1}{\\sigma^2}$.\n",
+    "Fortunately, the solution does not depend on $\\sigma$.\n",
+    "It follows that minimizing squared error\n",
+    "is equivalent to maximum likelihood estimation\n",
+    "of a linear model under the assumption of additive Gaussian noise.\n",
+    "\n",
+    "## From Linear Regression to Deep Networks\n",
+    "\n",
+    "So far we only talked about linear functions.\n",
+    "While neural networks cover a much richer family of models,\n",
+    "we can begin thinking of the linear model\n",
+    "as a neural network by expressing it in the language of neural networks.\n",
+    "To begin, let us start by rewriting things in a 'layer' notation.\n",
+    "\n",
+    "### Neural Network Diagram\n",
+    "\n",
+    "Deep learning practitioners like to draw diagrams\n",
+    "to visualize what is happening in their models.\n",
+    "In :numref:`fig_single_neuron`,\n",
+    "we depict our linear model as a neural network.\n",
+    "Note that these diagrams indicate the connectivity pattern\n",
+    "(here, each input is connected to the output)\n",
+    "but not the values taken by the weights or biases.\n",
+    "\n",
+    "![Linear regression is a single-layer neural network. ](../img/singleneuron.svg)\n",
```

Comment:
This one seemed not to be correctly rendered

Reply:
adding images!